enumerable_project.rb

```diff
@@ -0,0 +1,387 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+
+# rubocop:disable Metrics/PerceivedComplexity
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        yield(key[0].to_s, key[1])
+
+      end
+
+    end
+  end
+
+  def my_each_with_index
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i], i)
+      end
+
+    elsif self.class == Hash
+
+      i = 0
+
+      for key in self
+
+        yield([key[0].to_s, key[1]], i)
+
+        i+=1
+
+      end
+
+    end
+  end
+
+  def my_select
+    if self.class == Array
+
+      arr = []
+
+      0.upto(self.length-1) do |i|
+        arr << self[i] if yield(self[i])
+      end
+
+      return arr
+
+    elsif self.class == Hash
+
+      hash = {}
+
+      for key in self
+
+        hash[key[0]] = key[1] if yield(key[0], key[1])
+
+      end
+
+      return hash
+
+    end
+  end
+
+  def my_all?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false unless yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false unless yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
+
+  def my_none?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false if yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false if yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
+
+  def my_count
+    count = 0
+
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        count += 1 if yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        count += 1 if yield(key[0], key[1])
+
+      end
+
+    end
+
+    return count
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is always expecting a block
![image](https://user-images.githubusercontent.com/52209067/78953522-551f0200-7a96-11ea-9bb4-0bdb5fa58ac7.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78953555-7253d080-7a96-11ea-8aa1-33c6e62c768a.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)